syft/execution/plan.py

```diff
@@ -296,7 +296,10 @@ def __call__(self, *args):
                 args = (*args, self.state)
             return self.forward(*args)
         else:
-            return self.role.execute(args)
+            result = self.role.execute(args)
+            if len(result) == 1:
```

Comment:
I think the length of the result should always be `1` here (at least until we merge support for lists, etc.) Probably a nit: maybe we should raise an error if it isn't?

Reply:
Well, we do support several outputs for a plan already!
We only add a check to verify that everything is always the same on different runs in the other PR (I think)